### DISCUSSION TITLE
Array querystring output customization.

### DISCUSSION THREAD

**Discussion by sunwrobert** [47ng/nuqs#427] [Ideas] [answered]
Using Fastify on our backend and it supports queryparams in the form of 

`?foo=a&foo=b...` or `foo[0]=a&foo[1]=b`

https://github.com/fastify/fastify/issues/2841

Wondering if we could customize with not using a delimitted approach entirely? Or is there a way to do that right now.

**Comment by franky47** [comment, marked as answer]
It boils down to how `URLSearchParams` handle URLs with such patterns, and unfortunately it follows the [specification](https://url.spec.whatwg.org/#dom-urlsearchparams-get), which means it will only handle the first value for duplicated keys (the first example you gave).

For the second one, it would require a lot more processing of keys to recompose such patterns, and there isn't a standard behind it (that I know of).

Moreover, the delimiter approach uses fewer characters, which can be relevant as URL sizes are limited.

**Reply by cers** [reply]
I might very well be missing a lot of context here, but `URLSearchParams` does support multiple values for the same key:
```
new URLSearchParams('foo=42&foo=1764').getAll('foo')
// [ "42", "1764" ]
````

**Reply by franky47** [reply]
@cers It does, there's a discussion about it here: https://github.com/47ng/nuqs/discussions/484
